python/pyarrow/util.py

```diff
@@ -244,15 +244,15 @@ def download_tzdata_on_windows():
     tzdata_compressed = os.path.join(tzdata_path, "tzdata.tar.gz")
     os.makedirs(tzdata_path, exist_ok=True)
 
-    from urllib.request import urlopen
-    with urlopen('https://data.iana.org/time-zones/tzdata-latest.tar.gz') as response:
+    import requests
+    with requests.get('https://data.iana.org/time-zones/tzdata-latest.tar.gz') as response:  # noqa
```

Comment:
I am slightly concerned. This is a utility function that we provide to users and it's going to fail with and `ImportError` with this change as we don't have a dependency on requests.
Should we use the old behavior and only try to use requests in case it fails or probably better try to use requests and in case of `ImportError` try with the old urllib request?

Reply:
What do you think about c13eb9f0e395be813f6836c462ff0bed802877d4? Feel free to suggest alternate approaches.

Reply:
(sans the typo in the comment which I pushed another commit up for)

Reply:
Resolving as I think we've addressed this adequately. Thanks for catching this, as you were right!